### DISCUSSION TITLE
shokaX主题自带的markdown渲染器的代码块渲染问题

### DISCUSSION THREAD

**Discussion by nonhana** [theme-shoka-x/hexo-theme-shokaX#243] [Q&A] [answered]
当我和平常一样在博客上编辑markdown文章的时候，运行hexo s命令进行预览，发现有关于代码块的部分无法进行正常渲染。
![image](https://github.com/theme-shoka-x/hexo-theme-shokaX/assets/105428118/4cfe5250-dbfb-485b-9f54-538e0456d2b4)
![image](https://github.com/theme-shoka-x/hexo-theme-shokaX/assets/105428118/4bab1656-78db-4302-a0df-02aab7baade2)
可以看到，原本的代码跑到了最下面，然后原本的markdown文章内容被反而被放进代码块里面了。
好像是全部的\`\`\`<语言>\`\`\`的代码块全都会出现这样的问题，把<语言>给删掉之后是正常渲染。
我不清楚是这款主题原本的问题还是markdown渲染器的问题，抑或是我自己对这款主题使用不当的问题。希望能够及时得到回复！

**Comment by zkz098** [comment]
cc @D-Sketon 
可偶发性复现，但复现条件不明（好像出现页面`hexo cl`后会变化）
出现时伴随现象是footer会被异常渲染两次（文章内一次，文章外一次）最终生成页面和图2类似
推测和 prism 有关

如果方便请把这个md文件发送到`zkz20081204#126.com`（#替换为@），我看一下具体情况 @nonhana

**Reply by nonhana** [reply]
感谢回复！具体的md文件我已经通过邮件附上，请查收！
同时我也按照您的说法，进行了一次hexo cl清理，确实出现了新的渲染情况：
![image](https://github.com/theme-shoka-x/hexo-theme-shokaX/assets/105428118/c8bcf3f5-757b-4355-bf29-afabe4c832aa)
原本的代码框按钮好像消失了？

**Reply by zkz098** [reply]
先更新SXC然后重新安装主题试试
```bash
pnpm add shokax-cli@0.1.0 -g
SXC install shokaX
```

**Reply by zkz098** [reply]
先把`_config.yml`中`minify`下的`html`的`enable`调成false看看，如果没有改善再看下面的

方便的话把hexo环境打包发我邮箱里
我这边一直复现不了，感觉问题有点奇怪

**Comment by D-Sketon** [comment]
和 #220 报错相同

**Comment by zkz098** [comment]
问题在于 #220 和本讨论均无法稳定复现，下述两图footer的结构均正常：
![image](https://github.com/theme-shoka-x/hexo-theme-shokaX/assets/92242020/70c4d836-ed4e-40cd-a4c4-f3b4583cd4b4)
![image](https://github.com/theme-shoka-x/hexo-theme-shokaX/assets/92242020/4063fc60-2a19-4bf0-9aa1-7c3a75317cb9)
注：post-nav不显示为CDN侧防盗链，A图$$后的字符删去结果不变

@D-Sketon

**Reply by zkz098** [reply]
目前排查结论：
footer重复渲染为pjax所致（无需修复此问题，因为这是意料外的情况）
但也只有这一个结论了，我这边的所有环境都不能复现

**Reply by zkz098** [reply]
提出一个猜想：可能和minify-html有关

**Reply by D-Sketon** [reply]
确实和minify有关

**Reply by zkz098** [reply]
#220 在关闭 minify 后有改善吗 @D-Sketon

**Reply by D-Sketon** [reply]
有改善

**Reply by nonhana** [reply]
是的，应该是minify的问题，我也将我自己的minify关闭后正常渲染了

**Reply by zkz098** [reply, marked as answer]
那应该就是minify-html的上游问题，我换个稳定一点的html压缩器看看
